### DISCUSSION TITLE
Variable in a Collection request

### DISCUSSION THREAD

**Discussion by zane-shus** [jenssegers/laravel-mongodb#2539] [Q&A] [answered]
This is not a feature request of a bug.  My a lack of skill issue.

I have a laravel API app, and its meant to take a GET ID and use that in a match to get the individual item from Mongo.  All my static queries are working ok.

My code is:
```
 $value = (int)$sub_responseItem['value'];
 $page = Track::raw(function ($collection) {
      return $collection->aggregate(
       [['$match' => ['my_id' => $value ]]]
                                            );
                                        });
    $response = new MyCollection($page);
```

My error is:
```
message": "Undefined variable $value",
    "exception": "ErrorException",
    "file": "/var/www/html/app/Http/Controllers/api/v1/MyController.php",
    "line": 79,  (this is the line with $value on it)
```

If I change this ` [['$match' => ['my_id' => $value ]]]` to this ` [['$match' => ['my_id' => 1 ]]]`, it works ok.

And if i do a `var_dump($value);`  I get `(int)1`

Can anyone see what I am doing wrong?

**Comment by alcaeus** [comment, marked as answer]
Closures don't automatically capture the current scope, so you have two ways of fixing this. If using a long closure, make sure to bind any variables used within the closure with the `use` keyword:
```php
$page = Track::raw(
    function ($collection) use ($value) {
        return $collection->aggregate([['$match' => ['my_id' => $value ]]]);
    }
);
```

Alternatively, use a short closure which automatically captures the local scope:
```php
$page = Track::raw(
    fn ($collection) => 
        $collection->aggregate([['$match' => ['my_id' => $value ]]])
);
```

I'll assume that you changed the code for brevity in this issue, but for the sake of completeness I'll mention that the query you're running neither requires an `aggregate` command, nor does it need to run through `raw`. Instead, you can query for this directly:
```php
$page = Track::find('my_id', $value)->first();
```

**Comment by zane-shus** [comment]
that was it, thank you.. your a champion..

I guess its a related question.. What is a long vs short closure?  My google foo let me down in searching (I used short from the above).

**Comment by alcaeus** [comment]
The correct name for the feature would be [Arrow Function](https://www.php.net/manual/en/functions.arrow.php), which is a more concise syntax for [anonymous functions](https://www.php.net/manual/en/functions.anonymous.php) introduced in PHP 7.4.

**Reply by zane-shus** [reply]
this helped.. thank you
